File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.swift
class Solution {
    func maxProfit(_ prices: [Int]) -> Int {

        // var n = prices.count 
        // var profit = 0
        // var maxProfit = 0

        // for i in 0..<n-1{
        //     for j in i+1..<n{

        //         if(prices[j]>prices[i]){
        //             profit = prices[j] - prices[i]
        //             maxProfit = max(profit, maxProfit)
        //         }

        //     }
        // }

        // return maxProfit


        //keeptrack of global minimum

        var mini = Int.max
        var maxProfit = 0
        var profit = 0

        for price in prices{
            mini = min(mini,price)

            profit = price - mini

            maxProfit = max(maxProfit,profit)
        }

        return maxProfit

    }
}

File: Data Structures & Algorithms/max-water-container/submission-4.java
class Solution {
    public int maxArea(int[] heights) {

        int n = heights.length;

        int l =0, r = n-1;
        int maxArea = 0;


        while(l<r){
            int area = Math.min(heights[l],heights[r])*(r-l);

            System.out.print(area+" "+ l +" "+r+"\n");

            

            if(heights[l]<heights[r]){
                l++;
            }else{
                r--;
            }
            maxArea = Math.max(maxArea,area);

        }

        return maxArea;
        
    }
}
